### ISSUE TITLE
unresolved external symbol sqlite3_soci_error::result when linking soci_sqlite3_test.exe on VS2015 U3

### ISSUE THREAD

**Issue by sinall** [SOCI/soci#528]
> Link:
>   C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\link.exe /ERRORREPORT:QUEUE /OUT:"C:\Users\xxx\Document
>   s\Visual_Studio_2015\Projects\soci-build\bin\Debug\soci_sqlite3_test.exe" /INCREMENTAL /NOLOGO kernel32.lib user32.li
>   b gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib ..\..\..\lib\Debug\soc
>   i_sqlite3_3_2.lib C:\Users\xxx\Documents\Visual_Studio_2015\Projects\Test2\Debug\x86\sqlcipher.lib C:\Users\xxx
>   \Documents\Visual_Studio_2015\Projects\Test2\Debug\x86\sqlcipher.lib ..\..\..\lib\Debug\soci_core_3_2.lib /MANIFEST /
>   MANIFESTUAC:"level='asInvoker' uiAccess='false'" /manifest:embed /DEBUG /PDB:"C:/Users/xxx/Documents/Visual_Studio
>   _2015/Projects/soci-build/bin/Debug/soci_sqlite3_test.pdb" /SUBSYSTEM:CONSOLE /TLBID:1 /DYNAMICBASE /NXCOMPAT /IMPLIB
>   :"C:/Users/xxx/Documents/Visual_Studio_2015/Projects/soci-build/lib/Debug/soci_sqlite3_test.lib" /MACHINE:X86 /SAF
>   ESEH  /machine:X86 "soci_sqlite3_test.dir\Debug\test-sqlite3.obj"
> test-sqlite3.obj : error LNK2019: unresolved external symbol "public: int __thiscall soci::sqlite3_soci_error::result(v
> oid)const " (?result@sqlite3_soci_error@soci@@QBEHXZ) referenced in function __catch$?test6@@YAXXZ$0 [C:\Users\xxx\D
> ocuments\Visual_Studio_2015\Projects\soci-build\backends\sqlite3\test\soci_sqlite3_test.vcxproj]
> C:\Users\xxx\Documents\Visual_Studio_2015\Projects\soci-build\bin\Debug\soci_sqlite3_test.exe : fatal error LNK1120:
>  1 unresolved externals [C:\Users\xxx\Documents\Visual_Studio_2015\Projects\soci-build\backends\sqlite3\test\soci_sq
> lite3_test.vcxproj]
> 

Briefly, the linker complains unresolved external symbol "public: int __thiscall ci::sqlite3_soci_error::result(void)const " when linking soci_sqlite3_test.exe.

I find that neither libsoci_core_3_2.lib nore libsoci_sqlite3_3_2.lib is in the parameter list.

Any idea?

**Comment by sinall** [comment]
The above error occurs on 3.2 branch.

**Comment by EliasCode2015** [comment]
I have found that problem with ODBC linking. Luckily SOCI developers have uploaded an NUget package that you can download which takes care of that.

**Comment by mloskot** [comment]
I reproduced the problem:

```
2>------ Build started: Project: soci_sqlite3_test, Configuration: Debug Win32 ------
2>C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.10.25017\bin\HostX86\x86\link.exe /ERRORREPORT:PROMPT /OUT:"F:\dev\soci\src\build\bin\Debug\soci_sqlite3_test.exe" /INCREMENTAL /NOLOGO ..\..\..\lib\Debug\soci_sqlite3_3_2.lib "F:\dev\vcpkg\installed\x86-windows\debug\lib\sqlite3.lib" "F:\dev\vcpkg\installed\x86-windows\debug\lib\sqlite3.lib" ..\..\..\lib\Debug\soci_core_3_2.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /manifest:embed /DEBUG /PDB:"F:/dev/soci/src/build/bin/Debug/soci_sqlite3_test.pdb" /SUBSYSTEM:CONSOLE /TLBID:1 /DYNAMICBASE /NXCOMPAT /IMPLIB:"F:/dev/soci/src/build/lib/Debug/soci_sqlite3_test.lib" /MACHINE:X86 /SAFESEH  /machine:X86 "soci_sqlite3_test.dir\Debug\test-sqlite3.obj"
2>test-sqlite3.obj : error LNK2019: unresolved external symbol "public: int __thiscall soci::sqlite3_soci_error::result(void)const " (?result@sqlite3_soci_error@soci@@QBEHXZ) referenced in function __catch$?test6@@YAXXZ$0
2>F:\dev\soci\src\build\bin\Debug\soci_sqlite3_test.exe : fatal error LNK1120: 1 unresolved externals
2>Done building project "soci_sqlite3_test.vcxproj" -- FAILED.
```

I patched `release/3.2` branch, should be fixed now.
